File: app/src/main/java/com/commit451/gitlab/view/LabCoatSwipeRefreshLayout.java
package com.commit451.gitlab.view;

import android.content.Context;
import android.support.v4.widget.SwipeRefreshLayout;
import android.util.AttributeSet;

import com.commit451.gitlab.R;


/**
 * Just so that we do not have to keep setting the colors everywhere
 */
public class LabCoatSwipeRefreshLayout extends SwipeRefreshLayout {

    public LabCoatSwipeRefreshLayout(Context context) {
        super(context);
        init();
    }

    public LabCoatSwipeRefreshLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        int[] colors = getResources().getIntArray(R.array.cool_colors);
        setColorSchemeColors(colors);
    }
}

File: app/src/main/java/com/commit451/gitlab/view/LabCoatSwipeRefreshLayout.kt
package com.commit451.gitlab.view

import android.content.Context
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import android.util.AttributeSet

import com.commit451.gitlab.R


/**
 * Just so that we do not have to keep setting the colors everywhere
 */
class LabCoatSwipeRefreshLayout : androidx.swiperefreshlayout.widget.SwipeRefreshLayout {

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    private fun init() {
        val colors = resources.getIntArray(R.array.cool_colors)
        setColorSchemeColors(*colors)
    }
}
